[CDS-88516]: docs improvement for delegate OOM error (#4950)
harness/developer-hub: docs/troubleshooting/troubleshooting-nextgen.md

```
3. Apply harness-delegate.yaml again to restart the Kubernetes delegate (`kubectl apply -f harness-delegate.yaml`).

### Out-of-memory error (Java heap space)

The delegate throws an error indicating `java.lang.OutOfMemoryError` or that the delegate has run out of heap space.

#### Cause

The container has run out of memory or the delegate has exceeded heap space.

#### Solution
  

> Review the container's memory usage. Harness recommends that you use `Xms` and `Xmx` JVM arguments to adjust the heap size of the delegate. You can provide these arguments via the `JAVA_OPTS` environment variable when you set up the delegate.

Reviewer: Even though Xms and Xmx are still used, with our upgrades to JDK11 and JDK17 there are new options which are more appropriate for containerized apps like delegate, so I would recommend those as well (we use them by default instead of Xms & Xmx). 
Maybe it could read something like: 
Harness recommends that you use `Xms` and `Xmx` or  `MinRAMPercentage` and `MaxRAMPercentage` JVM arguments

```
3. Apply harness-delegate.yaml again to restart the Kubernetes delegate (`kubectl apply -f harness-delegate.yaml`).

> ### OutOfMemoryError (Java heap space)

Reviewer 2: ```suggestion
### Out-of-memory error (Java heap space)

The delegate throws an error indicating `java.lang.OutOfMemoryError` or that the delegate has run out of heap space.
```

> In case the delegate throws an error indicating `java.lang.OutOfMemoryError` or that the delegate has run out of heap space, please review the container's memory usage. It's recommended to use `Xms` and `Xmx` JVM arguments to adjust the heap size of the delegate. These arguments can be provided via the `JAVA_OPTS` environment variable, while setting up the delegate.

Reviewer 2: ```suggestion
#### Cause

The delegate has run out of heap space or the container is out of memory.

 #### Solution
  
Review the container's memory usage Harness recommends that you use `Xms` and `Xmx` JVM arguments to adjust the heap size of the delegate. You can provide these arguments via the `JAVA_OPTS` environment variable when setting up the delegate.
```
